### ISSUE TITLE
output not in range 0-1

### ISSUE THREAD

**Issue by mingkimon-dev** [poppuppy/alpha-free-matting#3]
Hi all,

Thank you for this wonderful work. I trained the model on 2,000 cat samples (image as input and the trimap as ground truth) for 100 epochs. The final loss curve looks like this:

<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/7b4e2d51-964b-4910-b427-b1e35f2b47cc" />

It seems the model converged well...(?) But when I tested the final checkpoint using run_one_image.py, I found that the output alpha matte values are not in the expected [0, 1] range, but instead fall between [0, 0.5]. I noticed that a sigmoid activation is used in the output, so I would expect the values to be within (0, 1).

Could you kindly help me understand what might be causing this issue?

Many many thanks! 😊

**Comment by poppuppy** [comment]
I think run_one_image.py would give a PIL image in [0,255]. Have you seen the output image, and is it normal?

**Comment by mingkimon-dev** [comment]
> I think run_one_image.py would give a PIL image in [0,255]. Have you seen the output image, and is it normal?

thank you for the reply! i tried with 800 samples again with same setting, and the output range is now fine in [0,1]! Maybe just the model  wasnt trained well...

but the result doesn't make sense yet 🤔, here's some examples from the training set in inferencing time

<img width="1171" alt="Image" src="https://github.com/user-attachments/assets/c958ff1a-57d9-425f-bc98-4e426649bb47" />

<img width="878" alt="Image" src="https://github.com/user-attachments/assets/8198a3f0-5b1b-45f0-b7e5-3f5dd3969380" />

and the loss curve for more info

<img width="1044" alt="Image" src="https://github.com/user-attachments/assets/2c4412d5-3145-41b8-8abf-da638ab4bf82" />

i read your paper again and seems like my model is still converging, guess i just reach the step i pointed to?
<img width="979" alt="Image" src="https://github.com/user-attachments/assets/b18957be-bff4-4bcb-800e-5f9cf1718a5d" />

**Comment by poppuppy** [comment]
There may be a possible missing point. May you check the pretrained checkpoint is correctly loaded?

**Comment by mingkimon-dev** [comment]
> There may be a possible missing point. May you check the pretrained checkpoint is correctly loaded?

yeah i redownload the dino_deitsmall16_pretrain.pth backbone and trained again, and results make more sense now. Thank you! 

<img width="1166" alt="Image" src="https://github.com/user-attachments/assets/698fe721-f62d-4028-8ea5-48858346a34b" />

<img width="874" alt="Image" src="https://github.com/user-attachments/assets/e2eaeb97-084a-4c60-aadd-6d98a1769e28" />

and i share my loss log down here for anyone want to retrain this model :) 

<img width="1044" alt="Image" src="https://github.com/user-attachments/assets/3ceff8ee-46ad-49e2-9736-f546d46e5edd" />
